MekHQ/src/mekhq/campaign/mission/resupplyAndCaches/Resupply.java

```diff
@@ -0,0 +1,2371 @@
+                    // at which point we'd just continue anyway, so we might as well just ignore the exception.
+                }
+            }
+        }
+
+        double averageTonnage = (unitTonnage / unitCount) / 2;
+
+        if (adHocResupply) {
+            averageTonnage *= Math.max(1, Math.floor((double) contract.getRequiredLances() / 3));
+        }
+
+        // We're wanting to round down to the nearest 5 tons.
+        int dropCount = (int) Math.max(1, Math.floor((double) contract.getRequiredLances() / 3));
+        return Math.floor(averageTonnage / 20.0) * 5.0 * dropCount;
+    }
+
+    /**
+     * Calculate the estimated cargo requirements for a given campaign and contract.
+     *
+     * @param campaign The campaign for which the cargo requirements are being estimated.
+     * @param contract The contract associated with the campaign.
+     * @return The estimated cargo requirements in tons as a formatted string, e.g., "50t".
+     */
+    public static String getEstimatedCargoRequirements(Campaign campaign, AtBContract contract) {
+        double baseCapacity = calculateTargetCargoTonnage(campaign, contract, false);
+
+        return (baseCapacity * CARGO_MULTIPLIER) + "t";
+    }
+
+    /**
+     * Calculate the negotiationModifier for the Resupply.
+     * <p>
+     * If the contract type is Guerrilla Warfare, then the flagged commander of the campaign
+     * will be set as the negotiator. For other contract types, this method searches for the best
+     * Logistics Administrator among campaign admins to represent as the negotiator. If one is found
+     * and their negotiation skill is determined, the negotiationModifier is calculated based on
+     * their skill level, with a max value capped at 3.
+     * <p>
+     * Resulting negotiationModifier is stored in {@code negotiationModifier} instance variable.
+     */
+    private void calculateNegotiationSkill() {
+        Person negotiator;
+        negotiatorSkill = NONE.ordinal();
+
+        if (contract.getContractType().isGuerrillaWarfare()) {
+            negotiator = campaign.getFlaggedCommander();
+        } else {
+            negotiator = null;
+
+            for (Person admin : campaign.getAdmins()) {
+                if (admin.getPrimaryRole().isAdministratorLogistics()
+                    || admin.getSecondaryRole().isAdministratorLogistics()) {
+                    if (negotiator == null
+                        || (admin.outRanksUsingSkillTiebreaker(campaign, negotiator))) {
+                        negotiator = admin;
+                    }
+                }
+            }
+        }
+
+        if (negotiator != null) {
+            Skill skill = negotiator.getSkill(SkillType.S_NEG);
+
+            if (skill != null) {
+                int skillLevel = skill.getFinalSkillValue();
+                negotiatorSkill = skill.getType().getExperienceLevel(skillLevel);
+            }
+        }
+    }
```

Comment:
Most of those tests can be moved to small functions that just return true or false, so this function becomes less indented.

Also, try to use Optionals more frequently in those situations, we already use null alot in the system, we may keep passing null forward, its fine, but optionals of things that can be null would more easily help work with some of the evaluations we have to do like in functions such as this.